include/oxen/quic/types.hpp

```diff
@@ -8,6 +8,25 @@
 
 namespace oxen::quic
 {
+    class Stream;
+    struct Address;
+
+    namespace concepts
+    {
+        template <typename T>
+        concept stream_derived_type = std::derived_from<T, Stream>;
+
+        template <typename T>
+        concept quic_address_type = std::derived_from<T, Address>;
+
+        template <typename T>
+        concept raw_sockaddr_type =
+                std::same_as<T, sockaddr> || std::same_as<T, sockaddr_in> || std::same_as<T, sockaddr_in6>;
+
+        template <typename T>
+        concept gtls_datum_type = std::same_as<T, gnutls_datum_t>;
```

Comment:
This concept doesn't seem very useful compared to just directly using `std::same_as` where used:

```c++
template <concepts::gtls_datum_type T>
operator const T*() const { return &x; }
```
versus:
```c++
template <std::same_as<gnutls_datum_t> T>
operator const T*() const { return &x; }
```